app-modules/report/src/Filament/Widgets/AssetsTable.php

```diff
@@ -0,0 +1,197 @@
+use Filament\Tables\Columns\TextColumn;
+use Filament\Tables\Filters\SelectFilter;
+use Filament\Tables\Table;
+use Filament\Widgets\TableWidget as BaseWidget;
+use Illuminate\Database\Eloquent\Builder;
+use Livewire\Attributes\On;
+
+class AssetsTable extends BaseWidget
+{
+    public string $cacheTag;
+
+    protected static ?string $heading = 'Assets';
+
+    protected static bool $isLazy = false;
+
+    protected static ?string $pollingInterval = null;
+
+    protected int | string | array $columnSpan = [
+        'sm' => 1,
+        'md' => 4,
+        'lg' => 4,
+    ];
+
+    public function mount(string $cacheTag): void
+    {
+        $this->cacheTag = $cacheTag;
+    }
+
+    #[On('refresh-widgets')]
+    public function refreshWidget(): void
+    {
+        $this->dispatch('$refresh');
+    }
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->query(
+                Asset::query()
+                    ->with(['type', 'status', 'location', 'maintenanceActivities', 'latestCheckOut'])
+                    ->orderBy('created_at', 'desc')
+            )
+            ->columns([
+                TextColumn::make('name')
+                    ->label('Name')
+                    ->searchable(),
+                TextColumn::make('type.name')
+                    ->label('Type')
+                    ->searchable(),
+                TextColumn::make('status.name')
+                    ->label('Status'),
+                TextColumn::make('deployment_status')
+                    ->label('Deployment')
+                    ->getStateUsing(function (Asset $record): string {
+                        $hasActiveMaintenance = $record->maintenanceActivities()
+                            ->whereNotIn('status', [
+                                MaintenanceActivityStatus::Completed,
+                                MaintenanceActivityStatus::Canceled,
+                            ])
+                            ->exists();
+
+                        if ($hasActiveMaintenance) {
+                            return 'Maintenance';
+                        }
+
+                        $isCheckedOut = $record->latestCheckOut && is_null($record->latestCheckOut->asset_check_in_id);
+
+                        if ($isCheckedOut) {
+                            return 'Checked Out';
+                        }
+
+                        return 'Bench Stock';
+                    })
+                    ->badge(),
+                TextColumn::make('purchase_date')
+                    ->label('Purchase Date')
+                    ->date(),
+                TextColumn::make('purchase_age')
+                    ->label('Device Age'),
+            ])
+            ->filters([
+                SelectFilter::make('name')
+                    ->label('Name')
+                    ->options(fn (): array => Asset::query()->orderBy('name')->limit(50)->pluck('name', 'name')->toArray())
+                    ->searchable()
```

Comment:
By providing a static array to options, and `searchable` with no arguments, the search here will only ever search on the 50 records returned in `options`. You will never see the other records in the system show up here.

You will need to pass a function to `searchable` to do a similar query as the one in option options but with the lowered search term included as a where like